tests/acceptance/test_embedded_study.py

```diff
@@ -13,3 +17,39 @@ def test_generate_dataset(study, database):
 
     assert len(results) == 1
     assert results[0]["year"] == "1943"
+
+
+def test_validate_dummy_data_happy_path(tmp_path):
+    dummy_data = "patient_id,year\n1,1971\n2,1992"
+    dataset_def_path = tmp_path / "dataset.py"
+    dummy_data_path = tmp_path / "dummy_data.csv"
+    dataset_path = tmp_path / "dataset.csv"
+    dataset_def_path.write_text(trivial_dataset_definition)
+    dummy_data_path.write_text(dummy_data)
+    _generate_dataset(dataset_def_path, dummy_data_path, dataset_path)
```

Comment:
I didn't think it would be needed anywhere else, so I didn't think it was worth refactoring the study fixture for it. I don't think the fixture itself is over-engineered: it's going to be re-used over multiple studies so it makes sense to wrap all that logic up.